client/extensions/woocommerce/app/orders/orders-list.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * External dependencies
+ */
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import React, { Component } from 'react';
+import { times } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FormInputCheckbox from 'components/forms/form-checkbox';
+import { fetchOrders } from 'woocommerce/state/sites/orders/actions';
+import { setCurrentPage } from 'woocommerce/state/ui/orders/actions';
+import {
+	areOrdersLoading,
+	areOrdersLoaded,
+	getOrders,
+	getTotalOrdersPages
+} from 'woocommerce/state/sites/orders/selectors';
+import { getOrdersCurrentPage } from 'woocommerce/state/ui/orders/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import Table from 'woocommerce/components/table';
+import TableRow from 'woocommerce/components/table/table-row';
+import TableItem from 'woocommerce/components/table/table-item';
+
+class Orders extends Component {
+	componentDidMount() {
+		const { siteId, currentPage } = this.props;
+
+		if ( siteId ) {
+			this.props.fetchOrders( siteId, currentPage );
+		}
+	}
+
+	componentWillReceiveProps( newProps ) {
+		if ( newProps.currentPage !== this.props.currentPage || newProps.siteId !== this.props.siteId ) {
+			this.props.fetchOrders( newProps.siteId, newProps.currentPage );
+		}
+	}
+
+	getOrderStatus = ( status ) => {
+		const { translate } = this.props;
+		const classes = `orders__item-status is-${ status }`;
+		switch ( status ) {
+			case 'pending':
+				return <span className={ classes }>{ translate( 'Pending payment' ) }</span>;
+			case 'processing':
+				return <span className={ classes }>{ translate( 'Processing' ) }</span>;
+			case 'on-hold':
+				return <span className={ classes }>{ translate( 'On hold' ) }</span>;
+			case 'completed':
+				return <span className={ classes }>{ translate( 'Completed' ) }</span>;
+			case 'cancelled':
+				return <span className={ classes }>{ translate( 'Cancelled' ) }</span>;
+			case 'refunded':
+				return <span className={ classes }>{ translate( 'Refunded' ) }</span>;
+			case 'failed':
+				return <span className={ classes }>{ translate( 'Failed' ) }</span>;
+		}
+	}
+
+	renderOrderItems = ( order, i ) => {
+		const { translate, moment, siteId } = this.props;
+		return (
+			<TableRow key={ i }>
+				<TableItem className="orders__table-checkbox">
+					<FormInputCheckbox aria-label={ translate( 'Select order %(order)s', {
```

Comment:
:information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.